src/loader/Loader.js

```diff
@@ -988,440 +1118,658 @@ Phaser.Loader.prototype = {
+            }
         };
 
-        this._xhr.onerror = function () {
-            return _this[onerror](index);
+        xhr.ontimeout = function () {
+            try {
+                return onerror.call(_this, file, xhr);
+            } catch (e) {
+                _this.asyncComplete(file, e.message || 'Exception');
+            }
         };
 
-        this._xhr.send();
+        xhr.onprogress = function() {};
+
+        xhr.onload = function () {
+            try {
+                return onload.call(_this, file, xhr);
+            } catch (e) {
+                _this.asyncComplete(file, e.message || 'Exception');
+            }
+        };
+
+        file.requestObject = xhr;
+        file.requestUrl = url;
+
+        //  Note: The xdr.send() call is wrapped in a timeout to prevent an issue with the interface where some requests are lost
+        //  if multiple XDomainRequests are being sent at the same time.
+        setTimeout(function () {
+            xhr.send();
+        }, 0);
 
     },
 
     /**
-    * Private method ONLY used by loader.
+    * Give a bunch of URLs, return the first URL that has an extension this device thinks it can play.
+    *
+    * It is assumed that the device can play "blob:" or "data:" URIs - There is no mime-type checking on data URIs.
     *
     * @method Phaser.Loader#getAudioURL
     * @private
-    * @param {array|string} urls - Either an array of audio file URLs or a string containing a single URL path.
+    * @param {object[]||string[]} urls - See {@link #audio} for format.
+    * @return {string} The URL to try and fetch; or null.
     */
     getAudioURL: function (urls) {
 
-        var extension;
-
-        if (typeof urls === 'string') { urls = [urls]; }
-
         for (var i = 0; i < urls.length; i++)
         {
-            extension = urls[i].toLowerCase();
+            var url = urls[i];
+            var audioType;
 
-            if (extension.substr(0,5) === "blob:")
+            if (url.uri) // {uri: .., type: ..} pair
             {
-                return urls[i];
+                url = url.uri;
+                audioType = url.type;
             }
+            else
+            {
+                // Assume direct-data URI can be played if not in a paired form; select immediately
+                if (url.indexOf("blob:") === 0 || url.indexOf("data:") === 0)
+                {
+                    return url;
+                }
 
-            extension = extension.substr((Math.max(0, extension.lastIndexOf(".")) || Infinity) + 1);
+                if (url.indexOf("?") >= 0) // Remove query from URL
+                {
+                    url = url.substr(0, url.indexOf("?"));
+                }
 
-            if (extension.indexOf("?") >= 0)
-            {
-                extension = extension.substr(0, extension.indexOf("?"));
+                var extension = url.substr((Math.max(0, extension.lastIndexOf(".")) || Infinity) + 1);
```

Comment:
@pnstickne  This line here will throw an error 'extension is undefined'.  I'm not sure it it's correct by I fixed it on my local copy with: 

var extension = url.substr((Math.max(0, url.lastIndexOf(".")) || Infinity) + 1);